cumulusci/tasks/bulkdata/tests/test_load.py

```diff
@@ -0,0 +1,918 @@
+
+        local_ids = ["1"]
+
+        step = MockBulkApiDmlOperation(
+            sobject="Contact",
+            operation=DataOperationType.UPDATE,
+            api_options={},
+            context=task,
+            fields=[],
+        )
+        step.results = [DataOperationResult(None, False, "message")]
+
+        mapping = {"table": "Account", "action": "update"}
+
+        with self.assertRaises(BulkDataException) as ex:
+            task._process_job_results(mapping, step, local_ids)
+
+        self.assertIn("Error on record with id", str(ex.exception))
+        self.assertIn("message", str(ex.exception))
+
+    def test_generate_results_id_map__success(self):
+        task = _make_task(
+            LoadData,
+            {"options": {"database_url": "sqlite://", "mapping": "mapping.yml"}},
+        )
+
+        step = mock.Mock()
+        step.get_results.return_value = iter(
+            [
+                DataOperationResult("001000000000000", True, None),
+                DataOperationResult("001000000000001", True, None),
+                DataOperationResult("001000000000002", True, None),
+            ]
+        )
+
+        generator = task._generate_results_id_map(
+            step, ["001000000000009", "001000000000010", "001000000000011"]
+        )
+
+        assert list(generator) == [
+            ("001000000000009", "001000000000000"),
+            ("001000000000010", "001000000000001"),
+            ("001000000000011", "001000000000002"),
+        ]
+
+    def test_generate_results_id_map__exception_failure(self):
+        task = _make_task(
+            LoadData,
+            {"options": {"database_url": "sqlite://", "mapping": "mapping.yml"}},
+        )
+
+        step = mock.Mock()
+        step.get_results.return_value = iter(
+            [
+                DataOperationResult("001000000000000", True, None),
+                DataOperationResult(None, False, "error"),
+                DataOperationResult("001000000000002", True, None),
+            ]
+        )
+
+        with self.assertRaises(BulkDataException) as ex:
+            list(
+                task._generate_results_id_map(
+                    step, ["001000000000009", "001000000000010", "001000000000011"]
+                )
+            )
+
+        self.assertIn("Error on record", str(ex.exception))
+        self.assertIn("001000000000010", str(ex.exception))
+
+    def test_generate_results_id_map__respects_silent_error_flag(self):
+        task = _make_task(
+            LoadData,
+            {
+                "options": {
+                    "ignore_row_errors": True,
+                    "database_url": "sqlite://",
+                    "mapping": "mapping.yml",
+                }
+            },
+        )
+
+        step = mock.Mock()
+        step.get_results.return_value = iter(
+            [
+                DataOperationResult("001000000000000", True, None),
+                DataOperationResult(None, False, None),
```

Comment:
@davidmreed  I know this is just a test case but for work I'm doing I need to know whether a  result row like this could ever really exist? i.e. Error==None and Success=False .

Reply:
@prescod  I'm not entirely sure.